(feat): Magento integration doc (#12766)
newrelic/docs-website: src/content/docs/apm/agents/php-agent/other-integrations/magento-business-insights.mdx

---
title: Magento business insights
tags:
    - New Relic integrations
    - Magento
metaDescription: Monitor your Magento application's performance with our comprehensive Magento business insights dashboard. Track key metrics like page load time, server response time, transaction time, error rate, and availability.
---

import apmMagentoDashboardOverview from 'images/apm_screenshot-full_magento-dashboard-overview.png'

Our Magento business insights dashboard provides a comprehensive view of your Magento ecommerce application's performance. With the dashboard, you can easily monitor key metrics like page load time, server response time, transaction time, error rate, and availability. You can also use the dashboard to diagnose issues in your app's performance and optimize your code.

<img
    title="Magento dashboard overview"
    alt="A screenshot of a dashboard with a variety of graphs and charts showing data about a Magento ecommerce application."
    src={apmMagentoDashboardOverview}
/>

<figcaption>

>     Our **Magento business insights** dashboard gives you a central view of your Magento app data. 

Author: ```suggestion
    Our Magento business insights dashboard gives you a central view of your Magento app data. 
```

---
title: Magento business insights
tags:
    - New Relic integrations
    - Magento

> metaDescription: Monitor your Magento application's performance with our comprehensive Magento business insights dashboard. Track key metrics like page load time, server response time, transaction time, error rate, and availability.

Author: ```suggestion
metaDescription: Monitor your Magento application's performance with our comprehensive Magento business insights dashboard. Track business metrics like revenue at risk, orders at risk due to downtime, and poor system performance.
```

---

import apmMagentoDashboardOverview from 'images/apm_screenshot-full_magento-dashboard-overview.png'

> Our Magento business insights dashboard provides a comprehensive view of your Magento ecommerce application's performance. With the dashboard, you can easily monitor key metrics like page load time, server response time, transaction time, error rate, and availability. You can also use the dashboard to diagnose issues in your app's performance and optimize your code.

Author: ```suggestion
Our Magento business insights dashboard provides a comprehensive view of your Magento ecommerce application's performance. With the dashboard, you can easily track key business metrics like revenue at risk, orders at risk due to downtime, and poor system performance. You can also use the dashboard to diagnose issues in your app's performance and optimize your code.
```

<img
    title="Magento dashboard overview"
    alt="A screenshot of a dashboard with a variety of graphs and charts showing data about a Magento ecommerce application."
    src={apmMagentoDashboardOverview}
/>

<figcaption>
    Our Magento business insights dashboard gives you a central view of your Magento app data. 
</figcaption>

To start monitoring your Magento app in New Relic, follow these steps to send your Magento app data to New Relic and view it in the Magento business insights dashboard.

> <Steps>

Author: ```suggestion

<Steps>
```

<Steps>

>     <Step>

Reviewer: is the infra agent optional? In most of the recent quickstart dashboard integrations, it's optional so just checking if that's true for this one

        ## Install the infrastructure agent

        Our infrastructure agent monitors your host and reports metrics such as CPU usage, memory usage, storage usage, disk usage, and network traffic. 
        

>         To install the agent, follow the steps in our [guided install](https://one.newrelic.com/launcher/nr1-core.explorer?pane=eyJuZXJkbGV0SWQiOiJucjEtY29yZS5saXN0aW5nIn0=&cards[0]=eyJuZXJkbGV0SWQiOiJucjEtaW5zdGFsbC1uZXdyZWxpYy5ucjEtaW5zdGFsbC1uZXdyZWxpYyIsImFjdGl2ZUNvbXBvbmVudCI6IlZUU09FbnZpcm9ubWVudCIsInBhdGgiOiJndWlkZWQifQ==).

Reviewer: i'm not sure if we should recommend the guided install. my understanding is that the guided install is a lightweight install approach that many customers don't use. I wonder if we should just say 'install the infra agent' and leave it at that, as they can decide based on the infra doc which route they want to go?

Reviewer: another option would be saying for each: "For a quick and easy way to install this, use our guided install. For a more manual and advanced install, see our install docs" or something similar.

        
        If you need to install the agent in a different way, see [Install the infrastructure monitoring agent](/docs/infrastructure/install-infrastructure-agent/get-started/install-infrastructure-agent/).

        </Step>

    <Step>

>         ## Install the PHP agent

Reviewer: these are missing the ## anchor text links (although I'm not sure if the Step formatting interferes with that or not)

        Our PHP agent reports application and browser-related metrics like AJAX performance, error rate, duration, core web vitals, and throughput.
        
        To install the agent, follow the steps in our [guided install](https://one.newrelic.com/launcher/nr1-core.settings?pane=eyJuZXJkbGV0SWQiOiJ0dWNzb24ucGxnLWluc3RydW1lbnQtZXZlcnl0aGluZyJ9&cards[0]=eyJuZXJkbGV0SWQiOiJzZXR1cC1uZXJkbGV0cy5zZXR1cC1waHAtaW50ZWdyYXRpb24iLCJhY2NvdW50SWQiOjI2NDA0MDl9&platform[accountId]=1).
        
        If you need to install the agent in a different way, see [PHP agent installation overview](/docs/apm/agents/php-agent/installation/php-agent-installation-overview/).

        </Step>
    
    <Step>
        ## Install the MySQL integration

>         Our MySQL integration monitors your databases and collects metrics like database utilization and connections such as slowest queries and active connections by node.

Reviewer: hard to parse this sentence, feels like it's got a wrong clause or is missing a comma, or something

        To install the integration:

            1. Go to **[one.newrelic.com](https://one.newrelic.com/) > Add data**.
            2. Search for and click **MySQL**.
            3. Select an account and follow the guided install steps.

> If you need to install the agent in a different way, see [MySQL monitoring integration](/install/mysql/).

Reviewer: ```suggestion
If you need to install the integration in a different way, see [MySQL monitoring integration](/install/mysql/).
```

        </Step>
    
    <Step>
        ## Install the Apache integration

        Our Apache integrations collects network metrics like TCP connections, DNS lookup, HTTPS, SSL, and server and worker statuses.

        To install the integration, follow the steps in our [guided install](https://one.newrelic.com/launcher/nr1-core.explorer?pane=eyJuZXJkbGV0SWQiOiJucjEtY29yZS5saXN0aW5nIn0=&cards[0]=eyJuZXJkbGV0SWQiOiJucjEtaW5zdGFsbC1uZXdyZWxpYy5ucjEtaW5zdGFsbC1uZXdyZWxpYyIsInBhdGgiOiJvaGkiLCJyZWNpcGVOYW1lIjoiYXBhY2hlLW9wZW4tc291cmNlLWludGVncmF0aW9uIiwiYWN0aXZlQ29tcG9uZW50IjoiVlRTT0NvbW1hbmQiLCJhY3RpdmVFbnZpcm9ubWVudCI6ImFwYWNoZS1vcGVuLXNvdXJjZS1pbnRlZ3JhdGlvbiJ9).
        

> If you need to install the agent in a different way, see [Apache monitoring integration](/docs/infrastructure/host-integrations/host-integrations-list/apache-monitoring-integration).

Reviewer: ```suggestion
If you need to install the integration in a different way, see [Apache monitoring integration](/docs/infrastructure/host-integrations/host-integrations-list/apache-monitoring-integration).
```

        </Step>
    

    <Step>
        ## Install the Elasticsearch integration
        

>         Our Elasticsearch integration collects metrics like cluster status and current shards. If you're using [Magento version 2.4 and higher](https://magentip.com/install-magento-2-4-x-on-ubuntu-with-elasticsearch/), you'll to install an ElasticSearch application and agent to monitor that application. 

Reviewer: "you'll to install" missing a word maybe

        
        To install the integration:

        1. Go to **[one.newrelic.com](https://one.newrelic.com/all-capabilities) > Add data**.
        2. Search for and click on **Elasticsearch**.
        3. Select an account and follow the guided install steps.

> If you need to install the agent in a different way, see [Apache monitoring integration](/docs/infrastructure/host-integrations/host-integrations-list/elasticsearch/elasticsearch-integration).

Reviewer: ```suggestion
If you need to install the integration in a different way, see [Elasticsearch monitoring integration](/docs/infrastructure/host-integrations/host-integrations-list/elasticsearch/elasticsearch-integration).
```

        </Step>
    

    <Step>
        ## Set up synthetic monitoring

        Our synthetic monitors test your app in different locations and capture server downtime. You can view this data in the Magento Business Insights dashboard.

        To set up synthetic monitors, follow the steps in our [guided install](https://onenr.io/0EjOXrnxgQ6).

        </Step>
    

    <Step>
        ## Install the AWS Identity and Access Management (IAM) integration
        
        Our AWS IAM integration collects metrics such as certificates per account, groups per account, MFA devices per account, roles per account, and users per account.

        To install the integration:

        1. Go to **[one.newrelic.com](https://one.newrelic.com/) > Add data**.
        2. Search for and click on **Integrate your AWS account**. 
        2. Select an account and follow the guided install steps.

For more information on this integration, see [AWS IAM integration](/docs/infrastructure/amazon-integrations/aws-integrations-list/aws-iam-monitoring-integration)
        </Step>
    

    <Step>
        ## Install the Flex integration

>         Our Flex integration allows you to capture business metrics through queries of your app database.

Reviewer: we might want to explain that the Flex integration isn't really a specific integration but something that allows you to make a custom integration/data-collection of your own. If I didn't know about it, I'd be a little confused thinking it was some specific kind of integration, not just a custom thing you set up.

                        - name: MYSQLTopTenSalesOrdersByValue
                        run: SELECT entity_id AS 'orderId', total_item_count AS 'orderQuantity', created_at AS 'createdDate', grand_total AS 'grandTotal' FROM sales_order ORDER BY grand_total DESC LIMIT 10;
                        - name: MYSQLOrdersAverage
                        run: SELECT AVG(order_count) AS 'AverageNumberOfOrders', Year AS 'Year' FROM (SELECT YEAR(created_at) AS 'Year', COUNT(DISTINCT order_id) AS order_count FROM sales_order_item GROUP BY MONTH(created_at), YEAR(created_at)) subquery GROUP BY Year;
                        - name: MYSQLProductsTotal
                        run: SELECT count(*) AS 'Total Products' FROM catalog_product_entity_varchar WHERE attribute_id = 73;
                        custom_attributes: # can apply additional at a nested level
                            attype: blocking-query
                            host: CustomMagentoMySQL
                ```

        3. [Restart the infrastructure agent](/docs/infrastructure/install-infrastructure-agent/manage-your-agent/start-stop-restart-infrastructure-agent/).
    
    </Step>

<Step>

>     ## Install the Magento business insights dashboard [#find-data]

Reviewer: we treat 'Magento Business Insights' as title case elsewhere, probably need to make it sentence case everywhere

        1. Go to **[one.newrelic.com](https://one.newrelic.com/all-capabilities) > Dashboards**.
        2. In the search bar, type "Magento business insights". The dashboard should appear. 
        3. Click on it to install.

Your Magento dashboard is considered a custom dashboard and can be found in the **Dashboards** UI. For information on using and editing dashboards, see [our dashboard docs](/docs/query-your-data/explore-query-data/dashboards/introduction-dashboards).

        </Step>
</Steps>

> ## Query your Magento data

Reviewer: add anchor link text for all headings

> Interested in querying your Magento app data? Here's an example NRQL query checking the total registered customers on your Magento platform:

Reviewer: for these kinds of integrations I wondered if a good boilerplate thing to say is something like "This integration reports data from all the agents and integrations involved in its install. For details on the data reported by those tools, see the docs for those tools. Here's an example of..."

        
If you need to install the agent in a different way, see [Apache monitoring integration](/docs/infrastructure/host-integrations/host-integrations-list/apache-monitoring-integration).

        </Step>
    

    <Step>
        ## Install the Elasticsearch integration
        
        Our Elasticsearch integration collects metrics like cluster status and current shards. If you're using [Magento version 2.4 and higher](https://magentip.com/install-magento-2-4-x-on-ubuntu-with-elasticsearch/), you'll to install an ElasticSearch application and agent to monitor that application. 
        
        To install the integration:

        1. Go to **[one.newrelic.com](https://one.newrelic.com/all-capabilities) > Add data**.
        2. Search for and click on **Elasticsearch**.
        3. Select an account and follow the guided install steps.

If you need to install the agent in a different way, see [Apache monitoring integration](/docs/infrastructure/host-integrations/host-integrations-list/elasticsearch/elasticsearch-integration).

        </Step>
    

    <Step>
        ## Create synthetic monitors

>         Our synthetic monitoring feature uses synthetic monitors to test your app in different geolocations and capture server downtime.

Author: @rahulbasu-nr Does synthetic monitor data populate one of the charts on the business insights dashboard?
